flycheck.el

```diff
@@ -6417,13 +6417,13 @@ about dogma."
     (nreverse errors)))
 
 (defun flycheck-elixir--check-for-dogma ()
-  "Check if the `dogma' subcommand is available.
+  "Check if `dogma' is installed and the file is saved.
 
-Check by parsing the output of `mix help'.
+Check by looking for deps/dogma.
 Used as a predicate for enabling the checker."
-  (let ((mix (flycheck-checker-executable 'elixir-dogma)))
-    (seq-find (lambda (l) (string-match-p "dogma" l))
-              (ignore-errors (process-lines mix)))))
+  (and
+   (flycheck-buffer-saved-p)
```

Comment:
I didn't know about this one until today. It's more compact and uses the flycheck format instead of json:

```
  (flycheck-define-checker elixir-dogma
    "Defines a checker for elixir with dogma"
    :command ("mix" "dogma" "--format" "flycheck" source)
    :predicate
    (lambda ()
      (file-exists-p "deps/dogma"))
    :error-patterns
    (
     (info line-start (file-name) ":" line ":" column ": " (or "C" "R" "D") ": " (optional (id (one-or-more (not (any ":")))) ": ") (message) line-end)
     (warning line-start (file-name) ":" line ":" column ": " (or "W" "E" "F") ": " (optional (id (one-or-more (not (any ":")))) ": ") (message) line-end))
    :working-directory
    (lambda (_checker)
      (locate-dominating-file buffer-file-name "mix.exs"))
    :modes (elixir-mode))
```

Let me know if I should just pull this one